File: 프로그래머스/0/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.java
class Solution {
    public String solution(String my_string, String alp) {
        // 정답 선언 및 초기화
        String answer = "";
        // alp의 대문자형인 alp2 선언 및 초기화 = toUpperCase() 사용
        String alp2 = alp.toUpperCase();
        // replace를 이용해 해당 문자를 대문자로 바꿈
        answer = my_string.replace(alp, alp2);
        
        // 정답 리턴
        return answer;
    }
}

File: 프로그래머스/0/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.py
def solution(my_string, alp):
    answer = ''
    i = 0
    
    while i < len(my_string):
        if my_string[i] == alp:
            answer += alp.upper()
            i += 1
        else:
            answer += my_string[i]
            i += 1
    
    return answer